bb_wrapper/wrapper/bb.py

```diff
@@ -101,8 +106,88 @@ def authenticate(self):
         }
         kwargs = dict(headers=header, verify=False, data=data)
 
-        response = requests.post(url, **kwargs)
-        response = self._process_response(response)
-        self.__access_token = response.data["access_token"]
-        self.__token_type = response.data["token_type"]
+        if self.__login is None:
+            self.__login = requests.post(url, **kwargs)
+            self.__login = self._process_response(self.__login)
+
+        self.__access_token = self.__login.data["access_token"]
+        self.__token_type = self.__login.data["token_type"]
+
+        return self.__login
+
+    def reauthenticate(self):
+        """
+        Reseta dados de login (access token e o tipo de token) e faz uma nova requisição de autenticação.
+        """
+        self.__login = None
+        self.authenticate()
+
+    def _delete(self, url, headers=None) -> requests.Response:
+        try:
+            self.authenticate()
+            response = super()._delete(url, headers)
+
+        except HTTPError as err:
+            if err.response.status_code in AuthorizationDeniedEnum.values():
+                self.reauthenticate()
+                response = super()._delete(url, headers)
+            else:
+                response = err.response
+
+        return response
```

Comment:
Esse tratamento de erros está meio estranho!

1. caso não seja um problema de autenticação, o erro precisa ser levantado adiante novamente!
2. pode ser utilizado recursão para fazer o retry da requisição original
3. está muito branda a verificação, qualquer 401 ou 403 tenta autenticar novamente

```suggestion
        except HTTPError as err:
            if err.response.status_code in AuthorizationDeniedEnum.values():
                self.reauthenticate()
                return = self._delete(url, headers)
            raise err

        return response
```